test/unit/org/apache/cassandra/cql3/CQLTester.java

```diff
@@ -1520,21 +1530,98 @@ protected ResultMessage.Prepared prepare(String query) throws Throwable
         return QueryProcessor.instance.prepare(formatQuery(query), ClientState.forInternalCalls());
     }
 
+    /**
+     * Enables external execution in {@link #execute(String, Object...)}, so queries get full validation and go
+     * through reconciliation.
+     */
+    protected static void enableExternalExecution()
+    {
+        externalExecution = true;
+    }
+
+    /**
+     * Disables external execution in {@link #execute(String, Object...)}, so queries won't get full validation nor
+     * go through reconciliation.
+     */
+    protected static void disableExternalExecution()
+    {
+        externalExecution = false;
+    }
+
+    /**
+     * Execute the specified query as either an internal query or an external query depending on the value of
+     * {@link #externalExecution}.
+     *
+     * @param query a CQL query
+     * @param values the values to bind to the query
+     * @return the query results
+     * @see #execute
+     * @see #executeInternal
+     */
     public UntypedResultSet execute(String query, Object... values)
     {
-        return executeFormattedQuery(formatQuery(query), values);
+        return externalExecution
+               ? executeExternal(query, values)
+               : executeInternal(query, values);
+    }
+
+    /**
+     * Execute the specified query as an internal query only for the local node. This will skip reconciliation and some
+     * validation.
+     * </p>
+     * For the particular case of {@code SELECT} queries using secondary indexes, the skipping of reconciliation means
+     * that the query {@link org.apache.cassandra.db.filter.RowFilter} might not be fully applied to the index results.
+     *
+     * @param query a CQL query
+     * @param values the values to bind to the query
+     * @return the query results
+     * @see CQLStatement#executeLocally
+     */
+    public UntypedResultSet executeInternal(String query, Object... values)
+    {
+        return executeFormattedQuery(formatQuery(query), false, values);
+    }
+
+    /**
+     * Execute the specified query as an external query meant for all the relevant nodes in the cluster, even if
+     * {@link CQLTester} tests are single-node. This won't skip reconciliation and will do full validation.
+     * </p>
+     * For the particular case of {@code SELECT} queries using secondary indexes, applying reconciliation means that the
+     * query {@link org.apache.cassandra.db.filter.RowFilter} will be fully applied to the index results.
+     *
+     * @param query a CQL query
+     * @param values the values to bind to the query
+     * @return the query results
+     * @see CQLStatement#execute
+     */
+    public UntypedResultSet executeExternal(String query, Object... values)
```

Comment:
I've wondered what to call this method. I initially tried `executeDistributed` because it more accurately emulated what happens in a real distributed environment, or at least in a real server. But that name might give the wrong impression that it will be run in a multi-node cluster, as if it was a dtest. Then I've swapped to `executeExternal`, meaning that it runs the query as it comes from the exterior of the db, in opposition to internal queries. I cannot think of a better name, but it might be a lack of imagination.

Reply:
`executeWithCoordinatorValidation` ? just throwing idea, but I am also super bad with names

Reply:
Thanks for the feedback. I think the most important part we are trying to achieve here is not that much the coordinator validation but going through the full reconciliation path. I think I'll go with `executeWithCoordinator` for now. I guess something like `executeAsCoordinator`, `executeInCoordinator` or `executeCoordinator` will do too?

Reply:
Agreed on all points and the new suggested names are all ok with me. Thanks